File: app/src/main/java/wangdaye/com/geometricweather/db/converters/WeatherCodeConverter.java
package wangdaye.com.geometricweather.db.converters;

import android.text.TextUtils;

import io.objectbox.converter.PropertyConverter;
import wangdaye.com.geometricweather.common.basic.models.weather.WeatherCode;

public class WeatherCodeConverter implements PropertyConverter<WeatherCode, String> {

    @Override
    public WeatherCode convertToEntityProperty(String databaseValue) {
        if (TextUtils.isEmpty(databaseValue)) {
            return null;
        } else {
            return WeatherCode.getInstance(databaseValue);
        }
    }

    @Override
    public String convertToDatabaseValue(WeatherCode entityProperty) {
        return entityProperty.getId();
    }
}

File: app/src/main/java/wangdaye/com/geometricweather/db/converters/WeatherCodeConverter.kt
package wangdaye.com.geometricweather.db.converters

import io.objectbox.converter.PropertyConverter
import wangdaye.com.geometricweather.common.basic.models.weather.WeatherCode
import wangdaye.com.geometricweather.common.basic.models.weather.WeatherCode.Companion.getInstance

class WeatherCodeConverter : PropertyConverter<WeatherCode?, String?> {
    override fun convertToEntityProperty(databaseValue: String?): WeatherCode? =
        if (databaseValue.isNullOrEmpty()) null else getInstance(databaseValue)

    override fun convertToDatabaseValue(entityProperty: WeatherCode?): String? =
        entityProperty?.id
}
